src/libraries/System.Security.Cryptography.Encoding/tests/PemEncodingTests.cs

```diff
@@ -0,0 +1,548 @@
+        }
+
+        [Fact]
+        public static void TryFind_True_FindsPemAfterPemWithInvalidBase64()
+        {
+            string content = @"
+-----BEGIN TEST-----
+$$$$
+-----END TEST-----
+-----BEGIN TEST2-----
+Zm9v
+-----END TEST2-----";
+            Assert.True(PemEncoding.TryFind(content, out PemFields fields));
+            Assert.Equal("TEST2", content[fields.Label]);
+            Assert.Equal("Zm9v", content[fields.Base64Data]);
+        }
+
+        [Fact]
+        public static void TryFind_True_FindsPemAfterPemWithInvalidLabel()
+        {
+            string content = @"
+-----BEGIN ------
+YmFy
+-----END ------
+-----BEGIN TEST2-----
+Zm9v
+-----END TEST2-----";
+            Assert.True(PemEncoding.TryFind(content, out PemFields fields));
+            Assert.Equal("TEST2", content[fields.Label]);
+            Assert.Equal("Zm9v", content[fields.Base64Data]);
+        }
+
+        [Fact]
+        public static void TryFind_False_Empty()
+        {
+            Assert.False(PemEncoding.TryFind(string.Empty, out _));
+        }
+
+        [Fact]
+        public static void TryFind_False_PostEbBeforePreEb()
+        {
+            string content = "-----END TEST-----\n-----BEGIN TEST-----\nZm9v";
+            Assert.False(PemEncoding.TryFind(content, out _));
+        }
+
+        [Theory]
+        [InlineData("\tOOPS")]
+        [InlineData(" OOPS")]
+        [InlineData("-OOPS")]
+        [InlineData("te\x7fst")]
+        [InlineData("te\x19st")]
+        public static void TryFind_False_InvalidLabel(string label)
+        {
+            string content = $"-----BEGIN {label}-----\nZm9v\n-----END {label}-----";
+            Assert.False(PemEncoding.TryFind(content, out _));
+        }
+
+        [Fact]
+        public static void TryFind_False_InvalidBase64()
+        {
+            string content = "-----BEGIN TEST-----\n$$$$\n-----END TEST-----";
+            Assert.False(PemEncoding.TryFind(content, out _));
+        }
+
+        [Fact]
+        public static void TryFind_False_PrecedingLinesAndSignificantCharsBeforePreeb()
+        {
+            string content = "boop\nbeep-----BEGIN TEST-----\nZm9v\n-----END TEST-----";
+            Assert.False(PemEncoding.TryFind(content, out _));
+        }
+
+        [Fact]
+        public static void TryFind_False_ContentOnPostEbLine()
+        {
+            string content = "-----BEGIN TEST-----\nZm9v\n-----END TEST-----boop";
+            Assert.False(PemEncoding.TryFind(content, out _));
+        }
+
+        [Fact]
+        public static void TryFind_False_NoPostEncapBoundary()
+        {
+            string content = "-----BEGIN TEST-----\nZm9v\n";
+            Assert.False(PemEncoding.TryFind(content, out _));
+        }
+
+        [Fact]
+        public static void TryFind_False_IncompletePostEncapBoundary()
+        {
+            string content = "-----BEGIN TEST-----\nZm9v\n-----END TEST";
```

Comment:
I don't think I saw a test, one way or the other, on mismatched labels between preeb and posteb.